controllers/storagecluster/provider_server_test.go

```diff
@@ -149,10 +149,68 @@ func TestOcsProviderServerEnsureCreated(t *testing.T) {
 		assert.NoError(t, r.Client.Get(context.TODO(), client.ObjectKeyFromObject(secret), secret))
 	})
 
-	t.Run("Ensure that Service is not created when AllowRemoteStorageConsumers is enabled and ProviderAPIServerServiceType is set to any other value than NodePort or LoadBalancer", func(t *testing.T) {
+	t.Run("Ensure that Deployment,Service,Secret is created when AllowRemoteStorageConsumers and ProviderAPIServerServiceType set to ClusterIP", func(t *testing.T) {
 
 		r, instance := createSetupForOcsProviderTest(t, true, corev1.ServiceTypeClusterIP)
 
+		obj := &ocsProviderServer{}
+		res, err := obj.ensureCreated(r, instance)
+		assert.NoError(t, err)
+		assert.False(t, res.IsZero())
+
+		// storagecluster controller waits for svc status to fetch the IP and it requeue
+		// as we are using a fake client and it does not fill the status automatically.
+		// update the required status field of the svc to overcome the failure and requeue.
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		err = r.Update(context.TODO(), service)
+		assert.NoError(t, err)
+
+		// call ensureCreated again after filling the status of svc, It will fail on deployment now
+		res, err = obj.ensureCreated(r, instance)
+		assert.NoError(t, err)
+		assert.False(t, res.IsZero())
+
+		// storagecluster controller waits for deployment status to fetch the replica count and it requeue
+		// as we are using a fake client and it does not fill the status automatically.
+		// update the required status field of the deployment to overcome the failure and requeue.
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		deployment.Status.AvailableReplicas = 1
+		err = r.Update(context.TODO(), deployment)
```

Comment:
@nb-ohad can you please change

```suggestion
		err = r.Status().Update(context.TODO(), deployment)
```